File: script/solutions/selection_sort_simple_sol.py
# Ploblem: https://www.acmicpc.net/problem/2752
# [Required] Use the Selection Sort to solve it.

# Solver: Seonghwan Lee
# Solved Date: 2021-02-22
# BigO: N^2

def swap (l, a, b):
    temp = l[a]
    l[a] = l[b]
    l[b] = temp


def selection_sort(n_list):
    n = len(n_list)
    for i in range(n):
        temp_min_pos = i
        for j in range(i, n):
            if n_list[temp_min_pos] > n_list[j]:
                temp_min_pos = j
        if temp_min_pos != i:
            swap(n_list, i, temp_min_pos)

    return n_list

if __name__ == '__main__':
    n_list = list(map(int,input().split()))
    print(' '.join(map(str,selection_sort(n_list))))

File: script/solutions/selection_sort_simple-sol.cpp
// Ploblem: [세수정렬] https://www.acmicpc.net/problem/2752
// [Required] Use the Selection Sort to solve it.

// Solver: Seonghwan Lee
// Solved Date: 2021-03-07
// BigO: N^2

#include <iostream>
#include <vector>

using namespace std;

void selection_sort(int n, vector<int> &seq){
    for (int i = 0; i < n; i ++){
        int temp_min_pos = i;
        for (int j = i; j < n; j++) if (seq[temp_min_pos] > seq[j]) temp_min_pos = j;
        if (temp_min_pos != i) iter_swap(seq.begin() + i, seq.begin() + temp_min_pos);
    }
}

int main(){
    vector<int> seq;

    for (int i = 0; i < 3; i++){
        int e;
        scanf("%d", &e);
        seq.push_back(e);
    }

    selection_sort(3, seq);

    for (int i = 0; i < 3; i++){
        printf("%d ", seq[i]);
    }
    return 0;
}
